### DISCUSSION TITLE
Redirect to homepage if page not found?

### DISCUSSION THREAD

**Discussion by botv** [vercel/next.js#16861] [Help]
I have a site with a home URL of /dashboard. If a user visits a page that doesn't exist (i.e. /asdf) I'd like to redirect them to /dashboard. Is there a way to do this besides adding the redirect in the `getServerSideProps` on a custom 404 page?

**Comment by page-source** [comment]
Did you any solution to this?

**Reply by adebiyial** [reply]
There's a solution for this @botv @page-source 

Create a `404.js` file in the `pages` directory with the snippet.

```
import { useRouter } from 'next/router';
import { useEffect } from 'react';

export default function Lost() {
  const router = useRouter();

  useEffect(() => {
    router.push('/');
  }, [router]);

  return null;
}
```

What's happening here is simple. The `404.js` page will handle not found pages automatically, but we're using the `router` to detect and redirect to the index page `/`. You can redirect to whatever you like.
